### DISCUSSION TITLE
Assert expression equality

### DISCUSSION THREAD

**Discussion by saschahofmann** [ibis-project/ibis#8446] [Q&A] [answered]
I am looking for a method to assert whether two Expression instances are equal (read: compile to the same SQL). 

Just asserting equality doesn't work ( I am not super familiar with the internals but I assume that this isn't surprising but for completeness:
```python
TABLE.order_by("id") == TABLE.order_by("id")
```
returns `False`).

So I was wondering whether there is a method that could do this.

Currently, I am using the result of `.compile()` but with every Ibis update there are some slight changes that require me to change a lot of SQL strings.

**Comment by cpcloud** [comment, marked as answer]
Thanks for opening a discussion.

In general I would recommend against depending on the specifics of the generated SQL if possible.

Details like quoting, or the specific functions used to implement a feature may change across minor releases.

Expressions have a public `equals` method that allows you to do the kind of comparison you're talking about:

```
In [9]: from ibis.interactive import *

In [10]: t = ex.penguins.fetch()

In [11]: t.order_by("island").equals(t.order_by("island"))
Out[11]: True

In [12]: t.island.equals(t.island)
Out[12]: True

In [13]: t.island.equals(t.bill_depth_mm)
Out[13]: False
```

**Reply by saschahofmann** [reply]
Ah I saw that but for some reason thought this would be just an alias for `__eq__`.  Amazing 👍
